lightning/src/ln/channelmanager.rs

```diff
@@ -1836,9 +1839,19 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref> ChannelMan
 		let mut channel_state = Some(self.channel_state.lock().unwrap());
 		let removed_source = channel_state.as_mut().unwrap().claimable_htlcs.remove(&(payment_hash, *payment_secret));
 		if let Some(mut sources) = removed_source {
+			assert!(!sources.is_empty());
+			let passes_value = if let &Some(ref data) = &sources[0].payment_data {
+				assert!(payment_secret.is_some());
+				if data.total_msat == expected_amount { true } else { false }
+			} else {
+				assert!(payment_secret.is_none());
+				false
+			};
```

Comment:
just to check understanding -- we fail HTLCs without a payment_secret because it's a required feature?

Reply:
Oops, thats a big confusing - if this bool is set to false in this block it triggers the classic check later. I moved the old check into this block to make it clearer.

Reply:
Heh, well it (obviously) doesn't build if I pull the later check up into this block lol.